ChatSecure/Classes/Model/Yap Storage/OTRXMPPRoom.swift

```diff
@@ -77,7 +77,14 @@ extension OTRXMPPRoom:OTRThreadOwner {
     }
     
     public func avatarImage() -> UIImage {
-        return OTRImages.avatarImage(withUniqueIdentifier: self.uniqueId, avatarData: nil, displayName: nil, username: self.threadName())
+        
+        var image:UIImage? = OTRImages.image(withIdentifier: self.uniqueId)
+        if image == nil {
+            // If not cached, generate a default image and store that.
+            image = OTRGroupAvatarGenerator.avatarImage(withUniqueIdentifier: self.uniqueId, width: 100, height: 100)
+            OTRImages.setImage(image, forIdentifier: self.uniqueId)
+        }
+        return image!
```

Comment:
This could be rewritten to avoid force-unwrapping:

```swift
if let image = OTRImages.image(withIdentifier: self.uniqueId) {
  return image
} else {
  let image = OTRGroupAvatarGenerator.avatarImage(withUniqueIdentifier: self.uniqueId, width: 100, height: 100)
  OTRImages.setImage(image, forIdentifier: self.uniqueId)
  return image
}